### PR TITLE
Add the PendingOrganizationUserRole model to enable the auto role assignment

### PR DESCRIPTION
**Pull request by schenedx**
## Description

This is the PR to add the PendingUserOrganizationGroup so we can pre-assign user role before the user is created in core.

Also added the django signal to automatically assign newly created user into the proper organization_group if we added data to the PendingUserOrganizationGroup table

@edx/masters-neem Please review

### CODE REVIEW

File: registrar/apps/core/signals.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+logger = getLogger(__name__)
+
+
+def handle_user_post_save(sender, **kwargs):  # pylint: disable=unused-argument
+    """
+    Handle User model changes - checks if pending organization user role record exists.
+    If there are pending organization user role exists, assign the user the new OrganizationGroup
+    """
+    user_instance = kwargs.get("instance", None)
```
Comment (iloveagent57) [inline registrar/apps/core/signals.py]: Nit: `None` is already the default returned by `dict.get()` when the item doesn't exist.

File: registrar/apps/core/signals.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    if user_instance is None:
+        logger.info('User is null when we handle the user save signal!')
```
Comment (iloveagent57) [inline registrar/apps/core/signals.py]: I'd argue that this is actually an error.

File: registrar/apps/core/signals.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+from registrar.apps.core.models import PendingUserOrganizationGroup
+
+logger = getLogger(__name__)
+
+
+def handle_user_post_save(sender, **kwargs):  # pylint: disable=unused-argument
+    """
+    Handle User model changes - checks if pending organization user role record exists.
+    If there are pending organization user role exists, assign the user the new OrganizationGroup
```
Comment (iloveagent57) [inline registrar/apps/core/signals.py]: "If a pending user organization group record exists, add the user to the OrganizationGroup and delete the pending record."

File: registrar/apps/core/models.py [hunk 4]
```diff
@@ -116,3 +115,38 @@ def save(self, *args, **kwargs):
...
         return 'OrganizationGroup: {} role={}'.format(self.organization.name, self.role)
+
+
+class PendingUserOrganizationGroup(TimeStampedModel):
+    """
+    Organization Membership model for user who have not yet been created in the user table
+
+    .. pii:: stores the user email address field for pending users.
+             The pii data gets deleted after a real user gets created
+    .. pii_types: email address
```
Comment (iloveagent57) [inline registrar/apps/core/models.py]: Ohh, try changing this to `email_address` (notice the underscore)

**Review by iloveagent57 — COMMENTED** [review]
I think the missing underscore is what's throwing off the pii checker.

File: registrar/apps/core/signals.py (hunk 1 above)
Comment (schenedx) [reply registrar/apps/core/signals.py]: Good point. Simplified

File: registrar/apps/core/signals.py (hunk 2 above)
Comment (schenedx) [reply registrar/apps/core/signals.py]: changed to `logger.error`

File: registrar/apps/core/signals.py (hunk 3 above)
Comment (schenedx) [reply registrar/apps/core/signals.py]: Updated

File: registrar/apps/core/models.py (hunk 4 above)
Comment (schenedx) [reply registrar/apps/core/models.py]: Thank you

**Review by iloveagent57 — APPROVED** [review]

**Comment by kdmccormick** [comment]
I realized #32 breaks the admin panel for adding `OrganizationGroups`. I fix it in #38. Nothing for you to do here other than rebase this once #38 merges, which will hopefully be in a moment.

File: registrar/apps/core/models.py [hunk 5]
```diff
@@ -116,3 +115,38 @@ def save(self, *args, **kwargs):
...
+    .. pii_types:: email_address
+    .. pii_retirement:: local_api
+    """
+    class Meta(object):
+        ordering = ['created']
+
+    user_email = models.EmailField()
+    organization_group = models.ForeignKey(OrganizationGroup)
+
+    unique_together = ('user_email', 'organization_group')
```
Comment (kdmccormick) [inline registrar/apps/core/models.py]: This should be within `Meta`

File: registrar/apps/core/permissions.py [hunk 6]
```diff
@@ -98,3 +98,9 @@ class OrganizationReadWriteEnrollmentsRole(OrganizationRole):
     OrganizationReadEnrollmentsRole,
     OrganizationReadWriteEnrollmentsRole
 ]
+
+ROLE_CHOICES = (
```
Comment (kdmccormick) [inline registrar/apps/core/permissions.py]: Please remove this; I think this was an artifact from rebasing. `ROLE_CHOICES` is now defined dynamically within `OrganizationGroup`.

**Review by kdmccormick — CHANGES_REQUESTED** [review]

File: registrar/apps/core/models.py (hunk 5 above)
Comment (schenedx) [reply registrar/apps/core/models.py]: This is done

File: registrar/apps/core/permissions.py (hunk 6 above)
Comment (schenedx) [reply registrar/apps/core/permissions.py]: Thank you. I forgot about the removal here.

**Review by kdmccormick — APPROVED** [review]
LGTM :+1:

**Pull request merged**